PulseEco/History and Forecast/CustomCalendar.swift

```diff
@@ -0,0 +1,313 @@
+//
+//  CustomCalendar.swift
+//  PulseEco
+//
+//  Created by Sara Karachanakova on 26.4.22.
+//  Copyright © 2022 Monika Dimitrova. All rights reserved.
+//
+
+import SwiftUI
+
+struct CustomCalendar: View {
+    
+    @State var currentDate: Date = Date()
+    @State var currentMonth: Int = 0
+    
+    @Binding var showingCalendar: Bool
+    @Binding var showPicker: Bool
+    
+    @State var selectedYear: Int = Calendar.current.component(.year, from: Date())
+    @State var selectedMonth: Int = Calendar.current.component(.month, from: Date()) - 1
+    
+    var calendar: Calendar = {
+        var cal = Calendar.current
+        cal.locale = Locale(identifier: Trema.appLanguageLocale)
+        return cal
+    }()
+    
+    let firstButtonColor = #colorLiteral(red: 0.05490196078, green: 0.03921568627, blue: 0.2666666667, alpha: 1)
+    let greenColor = #colorLiteral(red: 0.3035327792, green: 0.6464360356, blue: 0.4156317115, alpha: 1)
+    let orangeColor = #colorLiteral(red: 1, green: 0.5960784314, blue: 0, alpha: 1)
+    let redColor = #colorLiteral(red: 0.6941176471, green: 0.168627451, blue: 0.2196078431, alpha: 1)
+    let greyColor = #colorLiteral(red: 0, green: 0, blue: 0, alpha: 0.6)
+    let chevronColor = #colorLiteral(red: 0, green: 0, blue: 0, alpha: 0.54)
+    
+    var body: some View {
+        
+        VStack(spacing: 10) {
+            
+            if !showPicker {
+                calendarPicker
+            } else {
+                datePicker
+            }
+        }
+        .onChange(of: currentMonth) { newValue in
+            currentDate = getCurrentMonth()
+        }
+        .padding(.all)
+        .background(Color.white)
+    }
+    
+    @ViewBuilder
+    private func cardView(value: DateValueModel) -> some View {
+        
+        VStack {
+            
+            let flag = Calendar.current.isDate(value.date, equalTo: Date.now, toGranularity: .day)
```

Comment:
even though this flag is immediately used we should call it something descriptive